### PR TITLE
[WIP] helix-lsp code cleanup

### PR DESCRIPTION
**Pull request by batonius**
I've been looking at the `helix-lsp` crate and half-instinctively started to tidy it up, removed unused imports, deleted a copy of the `select_all` file from `futures`, got rid of a couple of allocations, renamed a few methods, etc.

I wonder if you would be willing to accept a cleanup/refactor PR with no fixes/new features in it? If so, I'll continue with `helix-lsp` in this PR, otherwise, feel free to close it :)

### CODE REVIEW

**Comment by archseer** [comment]
I'll accept some cleanups! That said, the `select_all` file was copied because we needed a change that needs to wait until futures 0.4: https://github.com/rust-lang/futures-rs/pull/2293

**Comment by batonius** [comment]
> That said, the `select_all` file was copied because we needed a change that needs to wait until futures 0.4: [rust-lang/futures-rs#2293](https://github.com/rust-lang/futures-rs/pull/2293)

That's what I figured out, but it seems like for now it's enough to borrow `self.incoming` mutably to get it working with the stock `select_all`.

**Comment by batonius** [comment]
I split off the `select_all` fix to #97 to make this PR a bit more focused.

**Comment by archseer** [comment]
Since you're working on a LSP cleanup, would you mind tackling error handling? #33 and #103 both fail because we don't gracefully return an error inside the registry (when calling `client.initialize()`)

**Comment by batonius** [comment]
Sure.

**Comment by batonius** [comment]
I implemented the fixes in #170, so this PR is mostly cosmetic.

I have my ideas about improving `helix-lsp` further, but I'll do it in a more focused way, so consider this PR ready for review.

File: helix-lsp/src/transport.rs [hunk 1]
```diff
@@ -26,153 +20,149 @@ pub enum Payload {
...
+                }
             }
-            headers.insert(parts[0].to_string(), parts[1].to_string());
         }
 
-        // find content-length
-        let content_length = headers.get("Content-Length").unwrap().parse().unwrap();
+        let content_length = content_length.unwrap();
 
+        //TODO: reuse vector
```
Comment (archseer) [inline helix-lsp/src/transport.rs:101]: 👍🏻

**Comment by archseer** [comment]
Thanks, the `lsp` definitely needs some work! I think the current callback system could be improved too, the `LspCallbacks` type sitting on `Application` is a bit ugly. If you're interested in further maintaining `helix-lsp` let me know, I'd be happy to discuss ideas on Matrix :)

**Pull request merged**